utils/task_tester/inginious-test-task

```diff
@@ -0,0 +1,374 @@
+        :param inputfiles:  All the submissions to re-execute for a given task.
+        :param task:        The task to test.
+        :post:              The list of failed submission test and the number of re-runned 
+                            submission have been updated.
+    """
+    result, grade, problems, tests, custom, state, archive, stdout, stderr = job_manager.new_job(0, task, data["input"], "Task tester", True)
+    job_done_callback({"result":result, "grade": grade, "problems": problems, "tests": tests, "custom": custom, "archive": archive, "stdout": stdout, "stderr": stderr}, filename, inputfiles, data)
+
+def test_task(course, taskid) -> tuple[list, int]:
+    """ Re-run submissions for a specific task.
+        :param course:  The course containing the task to test.
+        :param taskid:  The ID of the task to test.
+        :return:        The list of failed submissions and the number of submissions re-executed.
+        :post:          The containers for the list of failed submissions and the number of 
+                        re-executed submissions have been reset.
+    """
+
+    logger.info('-> Re-running submissions for task <%s>' % taskid)
+
+    """ Get task from its id """
+    task = course.get_task(taskid)
+
+    """ Build test directory path for current task """
+    test_dir = os.path.join(course.get_fs().prefix, taskid, 'test/')
+
+    """ List sample submissions for the current task """
+    inputfiles = glob.glob(test_dir + '*.test')
+
+    """ For each submission in the test directory, test the task with the specified input """
+    # TODO : create thread pool to parallelize the testing
+    for filename in inputfiles:
+        """ Open the input file and merge with limits """
+        if not os.path.exists(filename):
+            logger.warning('Submission file <%s> skipped because it does not seem to be reachable.')
+            continue
+            
+        with open(filename, 'r') as fd:
+            submission = inginious.common.custom_yaml.load(fd)
+        launch_job(filename, submission, inputfiles, task)
+    
+    result = (job_done_callback.failed, job_done_callback.jobs_done)
+
+    """ Simple reporting """
+    failed = len(job_done_callback.failed)
+    if  failed > 0:
+        logger.error('%i/%i tests failed\n%s' % (
+            failed,
+            job_done_callback.jobs_done,
+            '\n'.join(['- %s' % i for i in job_done_callback.failed])
+        ))
+    else:
+        if job_done_callback.jobs_done == 0:
+            logger.warning('No test submission found.')
+        else:
+            logger.success('-> All the previous submissions passed.')
+
+    """ Reset results containers """
+    job_done_callback.failed = []
+    job_done_callback.jobs_done = 0
+
+    return result
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(
+        'This tool replays submissions of a given course to ensure that its task\'s grading '
```

Comment:
Same comment here. A documentation page would be better to describe the tool beahviour in a comprehensive way, so this usage text can be reduced to the necessary.

Reply:
Done in https://github.com/UCL-INGI/INGInious/pull/861/commits/311e9e5f9ff24441f8ce3883f68d9d0cd89a5b63.